src/engine/guidance/lane_processing.cpp

```diff
@@ -27,9 +28,19 @@ std::vector<RouteStep> anticipateLaneChange(std::vector<RouteStep> steps,
 {
     // Lane anticipation works on contiguous ranges of quick steps that have lane information
     const auto is_quick_has_lanes = [&](const RouteStep &step) {
-        const auto is_quick = step.duration < min_duration_needed_for_lane_change;
         const auto has_lanes = step.intersections.front().lanes.lanes_in_turn > 0;
-        return has_lanes && is_quick;
+        if (!has_lanes)
+            return false;
+
+        // The more unused lanes to the left and right of the turn there are, the higher
+        // the chance the user is driving on one of those and has to cross lanes.
+        // Scale threshold for these cases to be adaptive to the situation's complexity.
+        const auto num_irrelevant = step.NumLanesToTheRight() + step.NumLanesToTheLeft();
```

Comment:
Yes you are right - but we don't know the lane source and target since we have no mapping.

Hm I assume a more conservative approach would be `max(lanes right, lanes left)`?